CommunityShed/CommunityShed/Item/ItemEdit.aspx.cs

```diff
@@ -0,0 +1,105 @@
+﻿using CommunityShed.Data;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace CommunityShed.Item
+{
+    public partial class ItemEdit : BasePage
+    {
+        int ItemId;
+        int CommunityId;
+        string redirect = "~/Default.aspx";
+
+        protected void Page_Init(object sender, EventArgs e)
+        {
+            if (!int.TryParse(Request.QueryString["CID"], out CommunityId))
+            {
+                Response.Redirect(redirect);
+            }
+
+            if (!CustomUser.IsInRole("Reviewer", CommunityId) && !isOwner())
+            {
+                Response.Redirect("~/RequestDenied.aspx");
+            }
+        }
+
+
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!int.TryParse(Request.QueryString["ID"], out ItemId))
+            {
+                Response.Redirect(redirect);
+            }
```

Comment:
I'd move this `if` statement to the `Page_Init` event handler, so that it's with the other query string parameter parsing bit.